ribs/archives/_cvt_archive.py

```diff
@@ -164,16 +170,59 @@ def __init__(self,
         self._chunk_size = chunk_size
 
         if custom_centroids is None:
-            if not isinstance(samples, int):
-                # Validate shape of custom samples. These are ignored when
-                # `custom_centroids` is provided.
-                samples = np.asarray(samples, dtype=self.dtype)
-                if samples.shape[1] != self._measure_dim:
-                    raise ValueError(
-                        f"Samples has shape {samples.shape} but must be of "
-                        f"shape (n_samples, len(ranges)={self._measure_dim})")
-            self._samples = samples
-            self._centroids = None
+            if centroid_method == "kmeans":
+                # Samples are set to custom first and then checked or used.
+                self._samples = samples
```

Comment:
Move this line to the end of the first if block below (just before the else). This will also ensure that self._samples is set to the numpy version of samples. I think your tests are currently failing because the tests pass in samples as a list, but it needs to be a numpy array (only numpy arrays have the tolist method). See the test failures for more info under the GitHub Actions.